### PR TITLE
Moved some chapters to pipeline extension

### PR DESCRIPTION
**Pull request by shuntrho**

### CODE REVIEW

File: basic_pipeline_extension/01_Introduction.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Introduction
+
+In this section, we gathered some topics which should give you better insight into Membrane. We will base on the [basic pipeline tutorial](../basic_pipeline/01_Introduction.md), so make sure you have gone through it first.
+The chapters are loosely related, so feel free to read them in any order you want.
```
Comment (varsill) [inline basic_pipeline_extension/01_Introduction.md:4]: How about listing the chapters of the tutorial here, and shortly describing (just like a single sentence) each of them?

**Review by varsill — COMMENTED** [review]
Great job! WDYT about adding the description of pipeline's extension topics, as I have asked in the comment? We can surely discuss it, but it is just a suggestion to write it, as I am not sure if we really need that description.

File: basic_pipeline_extension/01_Introduction.md (hunk 1 above)
Comment (shuntrho) [reply basic_pipeline_extension/01_Introduction.md:4]: This is a good idea!

**Review by varsill — APPROVED** [review]
Great! In fact, we are not following one of the rules from the contribution guide here (nesting of the headers should be done gradually), but as far as I see it looks much better if done this way. LGTM! 🥇

**Pull request merged**